annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+    assert doc_objs["doc1"]["objects"][1]["id"] == 102
+    assert doc_objs["doc1"]["objects"][2]["id"] == 103
+
+    mock_get_new_id.assert_called()
+    mock_get_common_ids.assert_called()
+    mock_get_links_and_children.assert_called()
+    mock_get_common_values.assert_called()
+
+
+def test_remove_ids():
+    task_obj = {
+        "id": 1,
+        "name": "Task 1",
+        "description": "This is a task.",
+        "links": [2, 3],
+        "children": [4, 5],
+        "extra_info": "Some extra info",
+    }
+
+    expected_result = {
+        "name": "Task 1",
+        "description": "This is a task.",
+        "extra_info": "Some extra info",
+    }
+
+    result = remove_ids(task_obj)
+
+    assert result == expected_result
+    assert "id" not in result
+    assert "links" not in result
+    assert "children" not in result
+
+
+@pytest.fixture
+def setup_data():
+    task_data = {
+        "objects": [
+            {"id": 1, "name": "Object A", "value": "Some Value", "links": [2]},
+            {"id": 2, "name": "Object B", "value": "Another Value"},
+        ]
+    }
+
+    all_tasks = {
+        1: [
+            ({"name": "Object A", "value": "Some Value"}, "1"),
+            ({"name": "Object C", "value": "Different Value"}, "3"),
+        ],
+        2: [({"name": "Object B", "value": "Another Value"}, "2")],
+    }
+
+    yield task_data, all_tasks
+
+
+def test_find_common_objs(setup_data: Dict[int, List[Tuple[Dict[str, str]]]]):
+    task_data, all_tasks = setup_data
+
+    expected_common_objs = [
+        {"id": 1, "name": "Object A", "value": "Some Value", "links": [2]},
+        {
+            "id": 2,
+            "name": "Object B",
+            "value": "Another Value",
+        },
+    ]
+
+    expected_same_ids = {1: ["1"], 2: ["2"]}
+
+    result_common_objs, result_same_ids = find_common_objs(
+        task_data, all_tasks
+    )
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
+
+
+def test_find_common_objs_no_common_objects():
+    task_data = {
+        "objects": [{"id": 3, "name": "Object D", "value": "Unique Value"}]
+    }
+
+    all_tasks = {
+        1: [({"name": "Object A", "value": "Some Value"}, "1")],
+        2: [({"name": "Object B", "value": "Another Value"}, "2")],
+    }
+
+    expected_common_objs = []
+    expected_same_ids = {}
+
+    result_common_objs, result_same_ids = find_common_objs(
+        task_data, all_tasks
+    )
+
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
+
+
+def test_find_common_objs_empty_data():
+    task_data = {"objects": []}
+
+    all_tasks = {}
+
+    expected_common_objs = []
+    expected_same_ids = {}
+
+    result_common_objs, result_same_ids = find_common_objs(
+        task_data, all_tasks
+    )
+
+    assert result_common_objs == expected_common_objs
+    assert result_same_ids == expected_same_ids
```

Comment:
These 3 tests use the same logic. Only one of them uses a fixture. We should use parametrized tests here, since it seems like it wouldn't add any extra logic.

Reply:
Yes, it can be done, but it might make the tests less readable due to their length.